python/MooseDocs/common/moose_docs_import.py

```diff
@@ -14,11 +14,30 @@
 ####################################################################################################
 #pylint: enable=missing-docstring
 import os
-import fnmatch
+import re
 import logging
 import MooseDocs
 LOG = logging.getLogger(__name__)
 
+def build_regex(pattern):
+    """
+    Build regex from paths with * and **.
+    """
+    out = pattern.replace('.', r'\.')
+    out = re.sub(r'(?!<^|/)(\*{2})(?=/|$)', r'(?:.*?)', out, flags=re.MULTILINE)
```

Comment:
I really think you should document these. This is pretty complex and will be difficult to maintain later. You can be pretty extreme with how you break them up: 
https://github.com/idaholab/moose/blob/devel/modules/chemical_reactions/src/actions/AddCoupledEqSpeciesAction.C#L51-L59